pkg/compiler/compiler_use_test.go

```diff
@@ -0,0 +1,121 @@
+package compiler_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/MontFerret/ferret/pkg/compiler"
+	"github.com/MontFerret/ferret/pkg/stdlib/strings"
+	"github.com/MontFerret/ferret/pkg/stdlib/types"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestUseExpression(t *testing.T) {
+	c := compiler.New()
+
+	c.Namespace("X").
+		RegisterFunction("INCLUDES", strings.Contains)
+	c.Namespace("Z").
+		RegisterFunction("STRING_FROM", types.ToString)
+	c.Namespace("Y").
+		RegisterFunction("INCLUDES", strings.Contains)
+
+	Convey("Use Expression", t, func() {
+
+		Convey("Should compile", func() {
+
+			Convey("Single statement", func() {
+				p, err := c.Compile(`
+				USE X
+	
+				RETURN INCLUDES("s", "s")
+				`)
+
+				So(err, ShouldBeNil)
+				out := p.MustRun(context.Background())
+
+				So(string(out), ShouldEqual, "true")
+			})
+
+			Convey("Many statements", func() {
+				p, err := c.Compile(`
+				USE X
+				USE Z
+	
+				RETURN STRING_FROM(INCLUDES("s", "s"))
+				`)
+
+				So(err, ShouldBeNil)
+				out := p.MustRun(context.Background())
+
+				So(string(out), ShouldEqual, `"true"`)
+			})
+
+			Convey("USE namespace twice", func() {
+				p, err := c.Compile(`
+				USE X
```

Comment:
Do you think we should allow this to compile?

Reply:
Think not. Will fix it